server/api/services/events.service.js

```diff
@@ -153,14 +153,18 @@ class EventsService {
       const points = {};
 
       winners.forEach((winner, index) => {
-        points[`score.${eventData.category}`] = eventData.importance * 5 + (4 - index) * 5;
-        points['totalScore'] = eventData.importance * 5 + (4 - index) * 5;
-
-        const achievement = {};
-        if (index === 0) achievement['achievements.first'] = eventData.name;
-        else if (index === 1) achievement['achievements.second'] = eventData.name;
-        else if (index === 2) achievement['achievements.third'] = eventData.name;
-
+        points[`score.${eventData.category}`] = eventData.importance * 5 + (2 - index) * 5;
+        points['totalScore'] = eventData.importance * 5 + (2 - index) * 5;
+
+        winner.forEach(userId => {
+          promises.push(
+            achievementModel.create({
+              userId: userId,
+              eventId,
+              position: index + 1,
+            })
+          )
+        });
         promises.push(
           userModel.updateMany(
```

Comment:
We'll have to remove the `$push: achievement` from here since we aren't pushing it in the users array anymore.